engine/src/main/java/org/pentaho/di/pan/PanCommandExecutor.java

```diff
@@ -326,10 +327,18 @@ public Trans loadTransFromRepository( Repository repository, String dirName, Str
     return trans; // return transformation loaded from the repo
   }
 
-  public Trans loadTransFromFilesystem( final String initialDir, final String filename, final String jarFilename ) throws Exception {
+  public Trans loadTransFromFilesystem( String initialDir, String filename, String jarFilename, String base64Zip )
+    throws Exception {
 
     Trans trans = null;
 
+    if ( !Utils.isEmpty( base64Zip ) ) {
+      //expected form of filename "*.ktr"
+      String zipName = java.util.UUID.randomUUID().toString() + ".zip";
+      filename = "zip:file:" + File.separator + File.separator + System.getProperty( "user.dir" ) + File.separator + zipName + "!" + filename;
```

Comment:
Use `Const.getUserHomeDirectory()` instead of `System.getProperty( "user.dir" )`

https://github.com/pentaho/pentaho-kettle/blob/master/core/src/main/java/org/pentaho/di/core/Const.java#L2054-L2057

Reply:
those are not equivalent, fyi, but I modified the code such that it will work